crates/solvers/src/api/mod.rs

```diff
@@ -46,3 +47,74 @@ impl Api {
         server.with_graceful_shutdown(shutdown).await
     }
 }
+
+// migrate to utoipauto once the issue is solved https://github.com/ProbablyClem/utoipauto/issues/23
+pub fn generate_openapi_yaml() -> Result<String, serde_yaml::Error> {
```

Comment:
Seems a bit backwards to generate this from the `solvers` repo. From my view the driver is the one dictating the interface so it makes more sense to me to derive the openapi spec from it's requests and expected responses.

Reply:
`gnosis/solvers` will generate the yaml file from the same crate. IMO, that should live where the API server with all the endpoints is declared.